### ISSUE TITLE
compute: end-to-end backpressure

### ISSUE THREAD

**Issue by teskje** [MaterializeInc/materialize#23897]

**Comment by teskje** [comment]
On the compute side we can just bring back the probe infrastructure removed in #22643. The biggest piece of the work will be in `persist_source` where the decision about when to emit how much data needs to be made.

**Comment by danhhz** [comment]
In a recent group DM, it seemed as if we were leaning toward the rate limiting working on _logical_ times rather than _physical_ characteristics like bytes.

**Comment by danhhz** [comment]
Yeah, that sounds great to me! If we want to speed up the timeline, we could also do it without the "After cutting the next release" step by including the plumbing in the cherry-pick part of the patch release, which doesn't feel particularly risky to me.

Also a note that the last time we did something like this, we didn't use the normal naming convention for the release tag. This made it very obvious that it shouldn't get accidentally deployed to other environments. I'd do the same here. https://github.com/MaterializeInc/materialize/releases/tag/v0.68.1-incident-70-0
